kubernetes/client.go

```diff
@@ -211,3 +211,26 @@ func filterPodsForService(selector labels.Selector, allPods *v1.PodList) []v1.Po
 	}
 	return pods
 }
+
+// filterPodsForEndpoints performs a second pass was selector may return too many data
+// This case happens when a "nil" selector (such as one of default/kubernetes service) is used
+func filterPodsForEndpoints(endpoints *v1.Endpoints, unfiltered *v1.PodList) []v1.Pod {
+	var pods []v1.Pod
+	for _, pod := range unfiltered.Items {
+		if hasEndpointForPod(endpoints, &pod) {
+			pods = append(pods, pod)
+		}
+	}
+	return pods
+}
+
+func hasEndpointForPod(endpoints *v1.Endpoints, pod *v1.Pod) bool {
```

Comment:
Suggestion, instead to iterate on every pod, you can create a map of bool with pod name.
Then you generate the map once and saves some iteration at the cost to maintain a map.

Reply:
done @lucasponce